cirq-core/cirq/transformers/gauge_compiling/gauge_compiling.py

```diff
@@ -0,0 +1,183 @@
+    """
+
+    def weight(self) -> float:
+        """Returns the relative frequency for selecting this gauge."""
+        return 1.0
+
+    @abc.abstractmethod
+    def sample(self, gate: ops.Gate, prng: np.random.Generator) -> "ConstantGauge":
+        """Returns a ConstantGauge sampled from a family of gauges.
+
+        Args:
+            gate: The two qubit gate to replace.
+            prng: A numpy random number generator.
+
+        Returns:
+            A ConstantGauge.
+        """
+
+
+@dataclass(frozen=True)
+class ConstantGauge(Gauge):
+    """A gauge that replaces a two qubit gate with a constant gauge."""
+
+    two_qubit_gate: ops.Gate
+    pre_q0: _SINGLE_QUBIT_GATES_T = None
+    pre_q1: _SINGLE_QUBIT_GATES_T = None
+    post_q0: _SINGLE_QUBIT_GATES_T = None
+    post_q1: _SINGLE_QUBIT_GATES_T = None
+
+    def sample(self, gate: ops.Gate, prng: np.random.Generator) -> "ConstantGauge":
+        return self
+
+    @functools.cached_property
+    def pre(self) -> Tuple[Sequence[ops.Gate], Sequence[ops.Gate]]:
+        """A tuple (ops to apply to q0, ops to apply to q1)."""
+        return _as_sequence(self.pre_q0), _as_sequence(self.pre_q1)
+
+    @functools.cached_property
+    def post(self) -> Tuple[Sequence[ops.Gate], Sequence[ops.Gate]]:
+        """A tuple (ops to apply to q0, ops to apply to q1)."""
+        return _as_sequence(self.post_q0), _as_sequence(self.post_q1)
+
+
+def _select(choices: Sequence[Gauge], probabilites: np.ndarray, prng: np.random.Generator) -> Gauge:
+    return choices[prng.choice(len(choices), p=probabilites)]
+
+
+@dataclass(frozen=True)
+class GaugeSelector:
+    """Samples a gauge from a list of gauges."""
+
+    gauges: Sequence[Gauge]
+
+    @functools.cached_property
+    def _weights(self) -> np.ndarray:
+        weights = np.array([g.weight() for g in self.gauges])
+        return weights / np.sum(weights)
+
+    def __call__(self, prng: np.random.Generator) -> Gauge:
+        """Randomly selects a gauge with probability proportional to its weight."""
+        return _select(self.gauges, self._weights, prng)
+
+
+@transformer_api.transformer
+class GaugeTransformer:
+    def __init__(
+        self,
+        # target can be either a specific gate, gatefamily or gateset
+        # which allows matching parametric gates.
+        target: Union[ops.Gate, ops.Gateset, ops.GateFamily],
+        gauge_selector: Callable[[np.random.Generator], Gauge],
+    ) -> None:
+        """Constructs a GaugeTransformer.
+
+        Args:
+            target: Target two-qubit gate, a gate-family or a gate-set of two-qubit gates.
+            gauge_selector: A callable that takes a numpy random number generator
+                as an argument and returns a Gauge.
+        """
+        if isinstance(target, ops.Gate):
+            self.target: Union[ops.GateFamily, ops.Gateset] = ops.GateFamily(target)
+        else:
+            self.target = target
+        self.gauge_selector = gauge_selector
```

Comment:
Nit to make it clear type specification if for both if branches.

```suggestion
        self.target: Union[ops.GateFamily, ops.Gateset] = (
            ops.GateFamily(target) if isinstance(target, ops.Gate) else target
        )
        self.gauge_selector = gauge_selector
```